gateway/gateway.go

```diff
@@ -603,33 +607,32 @@ func (gateway *HandleT) getJobDataFromRequest(req *webRequestT) (jobData *jobFro
 			return
 		}
 
+		eventContext, _ := toSet["context"].(map[string]interface{})
```

Comment:
if `context` is not present in `toSet`, I think this may cause an issue, because if the type assertion doesn't hold the nil value for a map would be nil.


```suggestion
		eventContext, ok := misc.MapLookup(toSet, "context").(map[string]interface{})
		if ok {
				further lookups
		}
```